pkg/diff2/ordering.go

```diff
@@ -0,0 +1,27 @@
+package diff2
+
+import (
+	"log"
+
+	"github.com/StackExchange/dnscontrol/v4/pkg/dnsgraph"
+	"github.com/StackExchange/dnscontrol/v4/pkg/dnssort"
+)
+
+func orderByDependencies(changes ChangeList) ChangeList {
+	if DisableOrdering {
+		log.Println("[Info: ordering of the changes has been disabled.]")
+		return changes
+	}
+
+	a := dnssort.SortUsingGraph(changes)
+
+	if len(a.UnresolvedRecords) > 0 {
+		log.Printf("Warning: Found unresolved records %v.\n"+
+			"This can indicate a circular dependency, please ensure all targets from given records exist and no circular dependencies exist in the changeset."+
+			"These unresolved records are still added as changes and pushed to the provider, but will cause issues if and when the provider checks the changes\n",
+			dnsgraph.GetRecordsNamesForGraphables(a.UnresolvedRecords),
```

Comment:
(sorry for the delayed response... I was traveling for a week. I'm on sabbatical until July 30, so dnscontrol is low priority right now)

Thanks for adding this explanation!

Now that I've had more time to think about it, I have some new thoughts! :)

The important thing here is that it is a warning, not an error.  It won't stop someone from getting their work done, though they may get an error from their provider.

I think this message should link to a page with more details.  Maybe `documentation/ordering.md`?  The page should give an example dnsconfig.js change that would product this error, and how to fix it.  This doc could also mention the `--disableordering` flag and request that people file a bug.

With that, I think we'll be ready to merge this!

Tom

Reply:
(No problem, and no need for apologies. We all are normal humans with lives. I am getting married in about 3 months, so my priorities are also a bit less on the open-source side than they normally would be. )

I did as you asked, added docs and linked in the warning to said docs. I also fixed the tests, appearently analyze_test and compareconfig_test rely on the same source records